gm4_sunken_treasure/data/gm4_sunken_treasure/functions/tick.mcfunction

```diff
@@ -1,5 +1,8 @@
-execute as @a[scores={gm4_st_sand=1..}] at @s if block ~ ~ ~ minecraft:water if block ~ ~1 ~ minecraft:water if block ~ ~2 ~ minecraft:water if block ~ ~3 ~ minecraft:water if block ~ ~4 ~ minecraft:water anchored eyes positioned ^ ^ ^2 at @e[type=minecraft:item,limit=1,distance=..3,nbt={Item:{id:"minecraft:sand"},Age:0s}] run function gm4_sunken_treasure:spawn_treasure
+execute as @a[scores={gm4_st_sand=1..}] at @s if predicate gm4_sunken_treasure:in_deep_water anchored eyes positioned ^ ^ ^2 at @e[type=minecraft:item,limit=1,distance=..2.5,nbt={Item:{id:"minecraft:sand"},Age:0s}] run function gm4_sunken_treasure:treasure_sand
+execute as @a[scores={gm4_st_gravel=1..}] at @s if predicate gm4_sunken_treasure:in_deep_water anchored eyes positioned ^ ^ ^2 at @e[type=minecraft:item,limit=1,distance=..2.5,nbt={Item:{id:"minecraft:gravel"},Age:0s}] run function gm4_sunken_treasure:treasure_gravel
+execute as @a[scores={gm4_st_gravel=1..}] at @s if predicate gm4_sunken_treasure:in_deep_water anchored eyes positioned ^ ^ ^2 at @e[type=minecraft:item,limit=1,distance=..2.5,nbt={Item:{id:"minecraft:flint"},Age:0s}] run function gm4_sunken_treasure:treasure_flint
```

Comment:
These lines should be combined into a function call that then checks for the item. This will reduce the amount of commands run per tick.